### PR TITLE
feat: add recurrences grouped by tax on total_details

### PR DESCRIPTION
**Pull request by mdnm**
Tackles this ticket: https://e-pilot.atlassian.net/jira/software/c/projects/ST/boards/180?selectedIssue=ST-1412

This commit introduces a new attribute to our total_details called recurrencesByTax. 

It will be used when the UI needs to display the taxes grouped by tax rate and recurrence:
image

### CODE REVIEW

File: src/__tests__/fixtures/pricing.results.ts [hunk 1]
```diff
@@ -375,6 +375,23 @@ export const resultsWhenNoPricesProvided = {
           },
         },
       ],
+      recurrencesByTax: [
+        {
+          amount_subtotal: 0,
```
Comment (paulofph) [inline src/__tests__/fixtures/pricing.results.ts:380]: I feel that we don't need all this data. Ex: `amount_total`, `amount_subtotal`, `amount_total_decimal`.

I would expose only the required data, while you were out we had some issues with big payloads in the order entity. We even cleaned some fields that we did not use, so I would keep the same approach. :)

Comment (mdnm) [reply src/__tests__/fixtures/pricing.results.ts:380]: Makes sense, I'll clean this up

Comment (mdnm) [reply src/__tests__/fixtures/pricing.results.ts:380]: Should be good now

**Review by paulofph — COMMENTED** [review]
Lgtm 🚀

**Review by paulofph — APPROVED** [review]

**Pull request merged**